ts/screens/portfolio/AddCardScreen.tsx

```diff
@@ -0,0 +1,256 @@
+import {
+  Button,
+  Col,
+  Content,
+  Grid,
+  Icon,
+  Input,
+  Item,
+  Text
+} from "native-base";
+import * as React from "react";
+import { FlatList, Image, View } from "react-native";
+import { NavigationScreenProp, NavigationState } from "react-navigation";
+
+import I18n from "../../i18n";
+import ROUTES from "../../navigation/routes";
+
+import { PortfolioStyles } from "../../components/styles";
+import {
+  CreditCardIcons,
+  CreditCardType
+} from "../../types/portfolio/CreditCardType";
+
+import { none, Option, some } from "fp-ts/lib/Option";
+
+type Props = Readonly<{
+  navigation: NavigationScreenProp<NavigationState>;
+}>;
+
+type State = Readonly<{
+  cardNumber: Option<string>;
+  expireDate: Option<string>;
+  secureCode: Option<string>;
+  cardHolder: Option<string>;
+}>;
+
+export class AddCardScreen extends React.Component<Props, State> {
+  public static navigationOptions = {
+    title: I18n.t("portfolio.addCardTitle"),
+    headerBackTitle: null
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      cardNumber: none,
+      expireDate: none,
+      secureCode: none,
+      cardHolder: none
+    };
+  }
+
+  public onExpireDateChange(value: string) {
+    this.changeExpireDateAppeareance(value);
+  }
+
+  public changeExpireDateAppeareance(value: string) {
+    const str = value.trim();
+    if (str.match(/^[\d\s]*$/)) {
+      const newString = str.replace(/\s/, "").match(/\d{1,2}/g);
+      if (newString !== null) {
+        const formattedText = newString.join("/");
+        this.setState({ expireDate: some(formattedText) });
+      }
+    }
+  }
+
+  public changeCardnumberAppeareance(value: string) {
```

Comment:
please add an explanatory comment for each complex method (this is hard to grasp : )